### PR TITLE
Re-adds audible screams (now in modular form!)

### PR DESCRIPTION
**Pull request by ModDownloading**
## About The Pull Request

Adds back audible screaming via the *scream emote without editing TG files!

The way this works is by simply creating a small file that overwrites the original scream emote and allows for it to be audible. I was initially doubting whether this would work but it runs as intended on the newest Fulpcode without any warnings or errors.

Voluntary screams produce sound for all races with actual scream sounds. Non-voluntary screams still produce sounds.

## Why It's Good For The Game

In general, I think that Fulpstation prefers *scream making sound (possibly because unlike TG, they've had audible screams in the recent past).

Before opening this pull request I asked in OOC whether people preferred the pre-codebase transfer *scream, which made sound, or the post-codebase transfer *scream, which didn't. I got a few people saying that they liked the old way better. When I asked if there were any people that didn't want screams to make sound, nobody answered.

I think this is enough to warrant me making a PR on the topic, though now that this is on GitHub it'll be easier for me to tell if my "testing-the-waters" check actually reflects what the playerbase wants.

## Changelog
:cl:
add: Using the *scream emote creates sound again.
/:cl:

### CODE REVIEW

**Comment by JohnFulpWillard** [comment]
Was it’s removal intentional by TG?

**Comment by ModDownloading** [comment]
Yes, it was, but Fulpstation has had voluntary screams produce noise much more recently and judging by an (admittedly brief) survey, it seems people might want a return to that.

I know how all new PRs have to be modular, so I did what I could to make sure this one doesn't go against established Fulpstation policy.

**Comment by JohnFulpWillard** [comment]
Went through TG, this is what they had to say on a PR when you tried to add it over there, about 15 days ago:

image
image

I honestly agree with them, we already have to deal with things like this
image
Adding another audible emote onto this would make this much worse.

**Comment by SgtHunk** [comment]
More audible emotes will just make things worse than they already are with the annoying spinflipclap spammers.
Also, the file isn't even included in the .dme

**Comment by Helianthus00** [comment]
I hate emote spam and entirely agree with John/Mothblocks here. Spamming emotes that make sounds are already bad, I don't want another.

**Comment by ModDownloading** [comment]
@Helianthus00 I was just making a PR of this since I figured Fulp had experience with audible screams far more recently than TG has, and as such there probably wouldn't be as many emote spammers (or at least admins would have more experience dealing with emote spammers). I'm sorry for making assumptions that this would be more palatable to this playerbase than others.

Would you reconsider the PR if I add a way to prevent emote spam of screams and possibly other emotes like clapping? The PR I gave to TG had the ability to set a cooldown timer for each emote, as well as a cooldown for the emote's audio (so, if the clapping cooldown was 2 seconds and audio was 10, then people could only clap every 2 seconds and the claps would only make sound every 10 seconds, for instance). 

If I code that kind of functionality into this PR, and put in what you and John think are reasonable cooldowns, ensuring emotes are abused even less than they are now, would that be acceptable?

**Comment by Helianthus00** [comment]
@ModDownloading I'll re-open this so you can give it a go. At the moment, there are no rules to prevent emote spamming, so this is just a code thing, which is why I was unwilling to consider it before. However, if you can add some reasonable cooldowns, I'll take it more seriously.

**Comment by ModDownloading** [comment]
@Helianthus00 Sure thing! What are some suggestions that you have for emote and emote sound cooldowns?

**Comment by JohnFulpWillard** [comment]
If you want to add cooldowns, TG might be willing to accept them, then we can add the sound here for *scream if TG doesnt after that.

image

**Comment by ModDownloading** [comment]
@JohnFulpWillard Will do, as soon as TG merges or declines my Mafia fix PR, I'll write up a emote sound cooldown PR and pitch it to them. 

What do you think reasonable cooldowns for emote sounds would be? I was thinking somewhere between 5-10 seconds for clapping and laughing, since deathgasp has a 15 second cooldown and it's way more obnoxious. I think those are the only two that actually make noise, but please correct me if I'm wrong, wouldn't want to miss anything.

**Comment by remove32** [comment]
something that I think would make this acceptable is if it wasn't voluntary.

If you get shot, your character might scream. Other things like that. *scream is annoying. Thank god we don't have *fart

**Comment by ModDownloading** [comment]
@remove32 It's already involuntary, and you do already scream sometimes when taking a lot of damage. I'm adding a voluntary option with a hefty cooldown (deathgasp makes a sound with a 15-second cooldown, I'm putting the same cooldown here). Now that TG merged my other PR, I can work on the only-cooldowns version there, and if/when it gets merged here I can safely integrate voluntary screaming with that cooldown.

**Comment by Nobelium-XVIII** [comment]
Macro time

**Comment by ModDownloading** [comment]
@JohnFulpWillard @Helianthus00 Ok my PR to add cooldowns is now up on TG! 5-second cooldowns for all generic sound-producing emotes that are voluntary are being pitched, though that's subject to change.

**Comment by Helianthus00** [comment]
Great, I appreciate you doing that. Once it's merged, ping either me or windarkata and we can see about merging your changes.

**Comment by ModDownloading** [comment]
@Helianthus00 They closed my PR on TG github, but don't worry, it's only due to documentation issues, not functionality. I'll fix up those parts in my code and hopefully they'll consider it again.

Also, these changes here will be irrelevant if the other PR gets merged, as I would just use my new audio_cooldown variable for scream sounds instead. If you want you can close this PR and I can open a new one if my PR goes through on TG.

**Comment by SgtHunk** [comment]
If the other PR gets merged, we'll just close this.
You also don't need to open a PR here if yours gets merged on TG. We do regular updates to get what TG has merged on their end. I actually plan on updating us soon to re-activate mapdiff, I was waiting for their bar stools PR to get merged.

**Comment by ModDownloading** [comment]
@Helianthus00 @JohnFulpWillard @SgtHunk 

My PR got merged on TG! (https://github.com/tgstation/tgstation/pull/59479)

I've changed this PR so that it adds a 15 second audio cooldown on the scream emote, and the scream emote can now produce sound voluntarily. The other PR has to be merged here before this one though, since the PR I merged upstream is what adds audio_cooldown as a var in the first place.

**Comment by Helianthus00** [comment]
That's great to hear. After discussion with the other admins, I have full intent on merging this once we get that TG update. In the meantime, you can just let this hang out. I'll keep it from auto-closing.

File: code/modules/mob/living/carbon/human/emote_fulp_edit.dm
Comment (Fikou) [inline code/modules/mob/living/carbon/human/emote_fulp_edit.dm:6]: dont need this if youre editing the emote i assume? these are on the base thing

Comment (ModDownloading) [reply code/modules/mob/living/carbon/human/emote_fulp_edit.dm:6]: I just want to make sure this doesn't ditch a lot of the code from the original scream.

Comment (Fikou) [reply code/modules/mob/living/carbon/human/emote_fulp_edit.dm:6]: but it will now. it will ditch any updates made to the original scream upstream

**Comment by Helianthus00** [comment]
After TGU was merged, the linter started failing. Fix it and we can merge this.

**Comment by ModDownloading** [comment]
I'll try to get the code working again. I'm not sure if they will accept this upstream considering I tried porting it there a month or two ago and they denied it.

**Comment by JohnFulpWillard** [comment]
> 
> 
> I'll try to get the code working again. I'm not sure if they will accept this upstream considering I tried porting it there a month or two ago and they denied it.

That was before the new PRs you added, which were accepted, of adding cooldowns to it.

File: code/modules/mob/living/carbon/human/emote_fulp_edit.dm
Comment (JohnFulpWillard) [inline code/modules/mob/living/carbon/human/emote_fulp_edit.dm:1]: Also this is in code/ rather than fulp_modules/

**Comment by ModDownloading** [comment]
I will move it to fulp_modules. Also, although my initial PR doesn't mention it, I did code a cooldown in the scream PR and it was still denied on TG. I think I talk about the cooldown in the comments of that commit, since it was something I added afterwards.

**Comment by ModDownloading** [comment]
Alright, moving the file into fulp_modules under a new folder emotes, changing the tgstation.dme and making the changes Fikou wanted. If everything works out I'll update the code here on Github to be identical and then see if that fixes the linters.

**Comment by ModDownloading** [comment]
@Helianthus00 @JohnFulpWillard Should be good to go! I tested it and it works as intended with the changes!

**Review by SgtHunk — CHANGES_REQUESTED** [review]
The features folder, is for, well, features. General overwrites to fit with our views go into the edits folder.

**Comment by ModDownloading** [comment]
@SgtHunk whoops! I'll move the file and reference in the dme right away, sorry about that.

File: fulp_modules/Z_edits/emote/fulp_emote_edits.dm
Comment (Fikou) [inline fulp_modules/Z_edits/emote/fulp_emote_edits.dm:3]: trailing newline

Comment (ModDownloading) [reply fulp_modules/Z_edits/emote/fulp_emote_edits.dm]: Fixed it, thanks.

**Comment by ModDownloading** [comment]
Should be good to go now, thanks for the help along the way.

**Comment by JohnFulpWillard** [comment]
image

**Pull request merged**
